File: Examples/VisualBasic/AsposePdfFacades/Stamps-Watermarks/ExtractImageImageStamp.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.StampsWatermarks
    Public Class ExtractImageImageStamp
        Public Shared Sub Run()
            Try
                ' ExStart:ExtractImageImageStamp
                ' The path to the documents directory.
                Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_StampsWatermarks()
                ' Instantiate PdfContentEditor Object
                Dim pdfContentEditor As New PdfContentEditor()

                ' Bind input PDF file
                pdfContentEditor.BindPdf(dataDir & Convert.ToString("ExtractImage-ImageStamp.pdf"))

                ' Get Stamp info for the first stamp
                Dim infos As StampInfo() = pdfContentEditor.GetStamps(1)

                ' Get the image from Stamp Info           
                Dim image As System.Drawing.Image = infos(0).Image

                ' Save the extracted image
                ' ExEnd:ExtractImageImageStamp  
                image.Save(dataDir & Convert.ToString("image_out_.jpg"))
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePdfFacades/Stamps-Watermarks/ExtractImageImageStamp.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace Aspose.Pdf.Examples.CSharp.AsposePDFFacades.StampsWatermarks
{
    public class ExtractImageImageStamp
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_StampsWatermarks();
            //Instantiate PdfContentEditor Object
            PdfContentEditor pdfContentEditor = new PdfContentEditor();

            ////bind input PDF file
            pdfContentEditor.BindPdf(dataDir+ "ExtractImage-ImageStamp.pdf");

            //Get Stamp info for the first stamp
            StampInfo[] infos = pdfContentEditor.GetStamps(1);

            //Get the image from Stamp Info
           
            System.Drawing.Image image = infos[0].Image;

            //Save the extracted image
            image.Save(dataDir+ "image.jpg");
 
            
            
        }
    }
}